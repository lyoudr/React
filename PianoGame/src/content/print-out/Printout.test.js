import React from 'react';
import PrintOut from './Printout';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

describe('pritOut test', () => {
  // Snapshot test
  it('renders correctly when searc', () => {
    const component = renderer.create(<PrintOut />);
    // 1. rendering
    let json_tree = component.toJSON();
    expect(json_tree).toMatchSnapshot();

    // manually trigger the callback
    component.getInstance().showLink('A');
    component.getInstance().toggleTheme();

    //2. re-rendering
    json_tree = component.toJSON();
    expect(json_tree).toMatchSnapshot();

    // manually trigger the callback
    component.getInstance().showLink('B');

    // 3. re-rendering
    json_tree = component.toJSON();
    expect(json_tree).toMatchSnapshot();
  });
  // DOM testing
  it('react testing library do testing', () => {
    const {container} = render(<PrintOut/>);
    const input = container.querySelector('#input');
    fireEvent.change(input, {target:{value: 'BACH'}});
    expect(input.value).toBe('BACH');
    setTimeout(() => {
      const bach_style = window.getComputedStyle(container.querySelector('#BACH'));
      const beethovan_style = window.getComputedStyle(container.querySelector('#Beethovan'));
      const mozart_style =  window.getComputedStyle(container.querySelector('#Mozart'));
      expect(bach_style.display).toBe('block');
      expect(beethovan_style.display).toBe('none');
      expect(mozart_style.display).toBe('none');
    }, 1500);
  });
});