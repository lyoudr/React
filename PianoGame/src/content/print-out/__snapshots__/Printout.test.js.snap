// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pritOut test renders correctly when searc 1`] = `
<div
  className="printout main"
>
  <section
    className="pieces"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 pb-5"
      >
        <input
          id="input"
          onChange={[Function]}
        />
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#eeeeee",
            }
          }
        >
          Search
        </button>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4 text-center pb-5 piano BACH show-on-scroll"
          id="BACH"
          onClick={[Function]}
        >
          <img
            src="pianosheet_1.gif"
          />
          <div
            className="mt-3"
          >
            BACH
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Beethovan show-on-scroll"
          id="Beethovan"
          onClick={[Function]}
        >
          <img
            src="pianosheet_2.gif"
          />
          <div
            className="mt-3"
          >
            Beethovan
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Mozart show-on-scroll"
          id="Mozart"
          onClick={[Function]}
        >
          <img
            src="pianosheet_3.gif"
          />
          <div
            className="mt-3"
          >
            Mozart
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="linkupload"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 link show-on-scroll"
      >
        <div
          className="col-md-8 text-center"
        >
          <iframe
            src=""
          />
          <h4>
            Vedio
          </h4>
        </div>
      </div>
      <div
        className="row justify-content-center mb-5 upload"
      >
        <div
          className="col-md-8 text-center"
        >
          <form>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlInput1"
              >
                Name
              </label>
              <input
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlYoutube"
              >
                Youtube Link
              </label>
              <input
                alt="youtube_link"
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlTextarea1"
              >
                Describe your pecies
              </label>
              <textarea
                className="form-control"
                rows="5"
              />
            </div>
          </form>
          <input
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <button
            className="btn btn-dark"
            onClick={[Function]}
            type="button"
          >
            Upload
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`pritOut test renders correctly when searc 2`] = `
<div
  className="printout main"
>
  <section
    className="pieces"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 pb-5"
      >
        <input
          id="input"
          onChange={[Function]}
        />
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#222222",
            }
          }
        >
          Search
        </button>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4 text-center pb-5 piano BACH show-on-scroll"
          id="BACH"
          onClick={[Function]}
        >
          <img
            src="pianosheet_1.gif"
          />
          <div
            className="mt-3"
          >
            BACH
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Beethovan show-on-scroll"
          id="Beethovan"
          onClick={[Function]}
        >
          <img
            src="pianosheet_2.gif"
          />
          <div
            className="mt-3"
          >
            Beethovan
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Mozart show-on-scroll"
          id="Mozart"
          onClick={[Function]}
        >
          <img
            src="pianosheet_3.gif"
          />
          <div
            className="mt-3"
          >
            Mozart
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="linkupload"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 link show-on-scroll"
      >
        <div
          className="col-md-8 text-center"
        >
          <iframe
            src=""
          />
          <h4>
            Vedio
          </h4>
        </div>
      </div>
      <div
        className="row justify-content-center mb-5 upload"
      >
        <div
          className="col-md-8 text-center"
        >
          <form>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlInput1"
              >
                Name
              </label>
              <input
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlYoutube"
              >
                Youtube Link
              </label>
              <input
                alt="youtube_link"
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlTextarea1"
              >
                Describe your pecies
              </label>
              <textarea
                className="form-control"
                rows="5"
              />
            </div>
          </form>
          <input
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <button
            className="btn btn-dark"
            onClick={[Function]}
            type="button"
          >
            Upload
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`pritOut test renders correctly when searc 3`] = `
<div
  className="printout main"
>
  <section
    className="pieces"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 pb-5"
      >
        <input
          id="input"
          onChange={[Function]}
        />
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#222222",
            }
          }
        >
          Search
        </button>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4 text-center pb-5 piano BACH show-on-scroll"
          id="BACH"
          onClick={[Function]}
        >
          <img
            src="pianosheet_1.gif"
          />
          <div
            className="mt-3"
          >
            BACH
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Beethovan show-on-scroll"
          id="Beethovan"
          onClick={[Function]}
        >
          <img
            src="pianosheet_2.gif"
          />
          <div
            className="mt-3"
          >
            Beethovan
          </div>
        </div>
        <div
          className="col-md-4 text-center pb-5 piano Mozart show-on-scroll"
          id="Mozart"
          onClick={[Function]}
        >
          <img
            src="pianosheet_3.gif"
          />
          <div
            className="mt-3"
          >
            Mozart
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="linkupload"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center mb-5 link show-on-scroll"
      >
        <div
          className="col-md-8 text-center"
        >
          <iframe
            src=""
          />
          <h4>
            Vedio
          </h4>
        </div>
      </div>
      <div
        className="row justify-content-center mb-5 upload"
      >
        <div
          className="col-md-8 text-center"
        >
          <form>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlInput1"
              >
                Name
              </label>
              <input
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlYoutube"
              >
                Youtube Link
              </label>
              <input
                alt="youtube_link"
                className="form-control"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="exampleFormControlTextarea1"
              >
                Describe your pecies
              </label>
              <textarea
                className="form-control"
                rows="5"
              />
            </div>
          </form>
          <input
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <button
            className="btn btn-dark"
            onClick={[Function]}
            type="button"
          >
            Upload
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
